library/common-library/DEVELOPERS.md

```diff
@@ -0,0 +1,615 @@
+{{ include "newrelic.common.privileged" . }}
+```
+
+### `newrelic.common.privileged.value`
+Returns true if privileged mode is enabled or false if not. This is to have the value of privileged ready to be templated.
+
+Usage:
+```mustache
+{{ include "newrelic.common.privileged.value" . }}
+```
+
+
+
+## _proxy.tpl
+### `newrelic.common.proxy`
+Returns the proxy URL configured by the user. 
+
+Usage:
+```mustache
+{{ include "newrelic.common.proxy" . }}
+```
+
+
+
+## _security-context.tpl
+These functions are to share a security context for all the charts in YAML format. Useful in clusters that have security
+enforced to not use the root user like OpenShift or users that had set an admission webhook.
+
+The functions are:
+* `newrelic.common.securityContext.container`
+* `newrelic.common.securityContext.pod`
+
+Usage:
+```mustache
+f and (.Values.controlPlane.enabled) (not (include "newrelic.fargate" .)) }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+    spec:
+      {{- with include "newrelic.common.securityContext.pod" . }}
+      securityContext:
+        {{- . | nindent 8 }}
+      {{- end }}
+
+      containers:
+        - name: example
+          {{- with include "nriKubernetes.securityContext.container" . }}
+          securityContext:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+```
+
+
+
+## _staging.tpl
+### `newrelic.common.nrStaging`
+Like almost everything in this library, it read global and local variables:
+```yaml
+global:
+  nrStaging:  # Note that this is empty (nil)
+nrStaging:  # Note that this is empty (nil)
+```
+
+Be careful: chart writers should NOT PUT ANY VALUE for this library to work properly. If in you
+values a `nrStaging` is defined, the global one is going to be always ignored.
+
+This function returns "true" of "" (empty string) so it can be used for evaluating conditionals.
+
+Usage:
+```mustache
+{{ include "newrelic.common.nrStaging" . }}
+```
+
+### `newrelic.common.nrStaging.value`
+Returns true if staging is enabled or false if not. This is to have the staging value ready to be templated.
+
+Usage:
+```mustache
+{{ include "newrelic.common.nrStaging.value" . }}
+```
+
+
+
+## _verbose-log.tpl
+### `newrelic.common.verboseLog`
+Like almost everything in this library, it read global and local variables:
+```yaml
+global:
+  verboseLog:  # Note that this is empty (nil)
+verboseLog:  # Note that this is empty (nil)
+```
+
+Be careful: chart writers should NOT PUT ANY VALUE for this library to work properly. If in you
+values a `verboseLog` is defined, the global one is going to be always ignored.
+
+Usage:
+```mustache
+{{ include "newrelic.common.verboseLog" . }}
+```
+
+### `newrelic.common.verboseLog.valueAsBoolean`
+Returns true if staging is enabled or false if not. This is to have the staging value ready to be templated as a boolean
+
+Usage:
+```mustache
+{{ include "newrelic.common.verboseLog.valueAsBoolean" . }}
+```
+
+### `newrelic.common.verboseLog.valueAsInt`
```

Comment:
copy-paste issue, verbose for Staging? Also below

Reply:
Fixed in [#6216d9f](https://github.com/newrelic/helm-charts/pull/797/commits/6216d9f3f09fedf3717c1b7a457a887fd8230371)